jobs_detector/main.py

```diff
@@ -1,33 +1,97 @@
-import click
-import requests
+import click, re, requests
 from bs4 import BeautifulSoup
+# from jobs_detector import settings
+import settings, exceptions
+from helpers import _keyword_search, _check_combinations, _get_percent
+from collections import defaultdict
 
-from jobs_detector import settings
-
-DEFAULT_KEYWORDS = [
-    # set some default keywords here
-]
+@jobs_detector.command()
+@click.option('-i', '--post-id', type=str, required=True, help= 'HN Post ID.')
+@click.option('-k', '--keywords', type=str, default=','.join(DEFAULT_KEYWORDS),
+help= "Specify keywords")
+@click.option('-c', '--combinations', type=str, help= "Specify keyword combinations to match",
+              callback=lambda _, x: x.split(',') if x else x)
+@click.argument('hacker_news')
 
 
 @click.group()
+
 def jobs_detector():
+    """
+    Find frequency of job-related keywords in a specified site.
+    Currently only supports hacker_news.
+    """
     pass
 
 
-@jobs_detector.command()
-@click.option('-i', '--post-id', type=str, required=True)
-@click.option('-k', '--keywords', type=str, default=','.join(DEFAULT_KEYWORDS))
-@click.option('-c', '--combinations', type=str,
-              callback=lambda _, x: x.split(',') if x else x)
+#Dict storage for keyword:num and keyword_list:[id1, id2] values
+keywords_dict = {}.setdefault
+comments_dict = {}
+DEFAULT_KEYWORDS = [
+    # set some default keywords here
+    'Remote',
+    'Postgres',
+    'Python',
+    'Javascript',
+    'React',
+    'Pandas',
+]
+
 def hacker_news(post_id, keywords, combinations):
     """
-    This subcommand aims to get jobs statistics by parsing "who is hiring?"
-    HackerNews posts based on given set of keywords.
+    Gets jobs statistics by searching HackerNews 
+    "who is hiring?" posts for given set of keywords.
     """
-    # HINT: You will probably want to use the `BeautifulSoup` tool to
-    # parse the HTML content of the website
-    pass
+    
+    combinations_list = combinations.split()
+    comments = []
+    combinations_dict = {}
+
+    # Fetch the post and make soup.
+    # BASE_URL = 'https://news.ycombinator.com/item?id={}'
+    url = BASE_URL.format(post_id)
+    r = requests.get(url)
+    page_content = r.content
+    soup = BeautifulSoup(page_content, 'html.parser')
+    
+    
+    for comment in soup.find_all('span', class_='c00'):
```

Comment:
You could also look only for parent comments by utilizing the `width` field of `img`, it will be >0 for replies to parent comments